sdk/device_update/src/client.rs

```diff
@@ -0,0 +1,209 @@
+use crate::Error;
+use azure_core::auth::{TokenCredential, TokenResponse};
+use const_format::formatcp;
+use url::Url;
+
+pub(crate) const API_VERSION: &str = "2021-06-01-preview";
+pub(crate) const API_VERSION_PARAM: &str = formatcp!("api-version={}", API_VERSION);
+
+/// Client for Device Update operations - import, list and delete updates
+///
+/// # Example
+///
+/// ```no_run
+/// use azure_device_update::DeviceUpdateClient;
+/// use azure_identity::token_credentials::DefaultAzureCredential;
+/// let creds = DefaultAzureCredential::default();
+/// let client = DeviceUpdateClient::new("contoso.api.adu.microsoft.com", &creds).unwrap();
+/// ```
+#[derive(Debug)]
+pub struct DeviceUpdateClient<'a, T> {
```

Comment:
We're generally avoiding tying clients to some specific lifetime. We should just make it easily cloneable.